conda/lock.py

```diff
@@ -49,6 +49,15 @@ def touch(file_name, times=None):
         os.utime(file_name, times)
 
 
+def preprocess_name(path):
```

Comment:
I think Locked (or the FileLock it turned into), might need another assert.  `assert not path.startswith('http')`

We also have utility functions to convert a `file://` url to a path.  I think it's `util.url_to_path`.  There's also a `connection.path_to_url` I think.  Someday we'll clean up where all this stuff is.